src/ILLink.RoslynAnalyzer/ISymbolExtensions.cs

```diff
@@ -22,6 +22,46 @@ internal static bool HasAttribute (this ISymbol symbol, string attributeName)
 			return false;
 		}
 
+		internal static bool TryGetDynamicallyAccessedMemberTypes (this ISymbol symbol, out DynamicallyAccessedMemberTypes? dynamicallyAccessedMemberTypes)
+		{
+			dynamicallyAccessedMemberTypes = null;
+			if (!symbol.HasAttribute (DynamicallyAccessedMembersAnalyzer.DynamicallyAccessedMembersAttribute))
+				return false;
+
+			var damAttributeName = DynamicallyAccessedMembersAnalyzer.DynamicallyAccessedMembersAttribute;
+			AttributeData? dynamicallyAccessedMembers = null;
+			foreach (var _attribute in symbol.GetAttributes ())
+				if (_attribute.AttributeClass is var attrClass && attrClass != null &&
```

Comment:
Can the attrClass ever be null? Just curious how that could happen.

Can this null check instead be `_attribute.AttributeClass?.HasName (damAttributeName) == true`?

The `HasName` check looks redundant with the `HasAttribute` check - could those be combined? Also should we include the namespace in `damAttributeName`?

Reply:
Added `TryGetAttribute` to get a named attribute on a symbol and simplify with your suggestions. This way we avoid unnecessary null checks and don't do the redundant `HasName` checking.